llm-chain-local/src/executor.rs

```diff
@@ -0,0 +1,195 @@
+use std::path::Path;
+
+use crate::options::{PerExecutor, PerInvocation};
+use crate::output::Output;
+use crate::LocalLlmTextSplitter;
+
+use async_trait::async_trait;
+use llm_chain::prompt::Prompt;
+use llm_chain::tokens::{PromptTokensError, Tokenizer, TokenizerError};
+use llm_chain::traits::ExecutorCreationError;
+
+/// Executor is responsible for running the LLM and managing its context.
+pub struct Executor {
+    llm: Box<dyn llm::Model>,
+    callback: Option<fn(&Output)>,
```

Comment:
What's the callback for no need for it methinks :)

Reply:
Indeed 😅